# DP: Assume working semaphores on Linux, don't rely on running kernel for the check.

Index: python2.7/configure.ac
===================================================================
--- python2.7.orig/configure.ac
+++ python2.7/configure.ac
@@ -4064,6 +4064,11 @@ int main(void) {
 [ac_cv_posix_semaphores_enabled=no],
 [ac_cv_posix_semaphores_enabled=yes])
 )
+case $ac_sys_system in
+  Linux*)
+    # assume enabled, see https://launchpad.net/bugs/630511
+    ac_cv_posix_semaphores_enabled=yes
+esac
 AC_MSG_RESULT($ac_cv_posix_semaphores_enabled)
 if test $ac_cv_posix_semaphores_enabled = no
 then
@@ -4100,6 +4105,11 @@ int main(void){
 [ac_cv_broken_sem_getvalue=yes],
 [ac_cv_broken_sem_getvalue=yes])
 )
+case $ac_sys_system in
+  Linux*)
+    # assume enabled, see https://launchpad.net/bugs/630511
+    ac_cv_broken_sem_getvalue=no
+esac
 AC_MSG_RESULT($ac_cv_broken_sem_getvalue)
 if test $ac_cv_broken_sem_getvalue = yes
 then
